product_tier_validation/models/product_state.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductState(models.Model):
+    _inherit = "product.state"
+
+    state = fields.Selection(
```

Comment:
@patrickrwilson Could you add a help in order to 'help' people to understand what is this for? This is one more layer and can introduce confusion in people mind...

IMHO, I'm not sure about this

Reply:
I can add a help, the reason we can't use 'code' is because of several reasons...
Tier validation needs 'to' and 'from' states in order to compute the validation needs so it needs to be in a selection field. 
The code field is user defined and unique so we can't really use it for the tier validation purpose.

Reply:
IMHO, definetely this is a bad implementation. A state... in a state. Quite insane.

Reply:
@kittiu What do you think of this ?

Reply:
@rousseldenis Indeed, the Model should have been named ProductStage.
"Stages" are configurable, while "States" are a predefined fixed list.
For business logic, we want "Stages" to be mapped to known "States" that can be safely used in business logic.